src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -1525,148 +1525,307 @@ static LogicalResult getPartiallyFlattenedSimdCode(
 // Utilities for Op fusion at lowering
 //===----------------------------------------------------------------------===//
 
-// Function pointer type for emitScalarOpFor<T>
+// Function pointer type for the emitScalarOpFor<T> of elementwise Ops.
 typedef mlir::Value (*EmitScalarFunc)(mlir::ConversionPatternRewriter &rewriter,
     mlir::Location loc, mlir::Operation *op, mlir::Type elementType,
     mlir::ArrayRef<mlir::Value> scalarOperands);
 
-// Variadic template to iterate all the fusible Ops
+// Utility class for Op fusion.
+// Start from the root op, which is being lowered as Elementwise Op.
+// A linear sequence of fusible ops are idenitified. The operations of these ops
+// are fused into the loop nest for the root Op, and the last op is replaced
+// and the others are deleted.
+// ToFix: fusion for a graph could be done in future.
+class OpFusionHelper {
+private:
+  mlir::Operation *op_;
+  mlir::ConversionPatternRewriter &rewriter_;
```

Comment:
I started with the declaration of class in a separate .hpp file and the `mlir::` is needed. Since the Elementwise templates are not in .hpp file, I have to keep these code in Elementwise.cpp now. However, these code eventually will be moved into a separate .hpp file.